pkg/api/quota_test.go

```diff
@@ -0,0 +1,125 @@
+	hs.Cfg.Quota.Org = &testOrgQuota
+	// Required while sqlstore quota.go relies on setting global variables
+	setting.Quota = hs.Cfg.Quota
+
+	_, err := hs.SQLStore.CreateOrgWithMember("TestOrg", testUserID)
+	require.NoError(t, err)
+
+	t.Run("AccessControl allows viewing CurrentOrgQuotas with correct permissions", func(t *testing.T) {
+		setAccessControlPermissions(acmock, []*accesscontrol.Permission{{Action: ActionOrgsRead, Scope: ScopeOrgsAll}})
+		response := callAPI(server, http.MethodGet, getCurrentOrgQuotasUrl, nil, t)
+		assert.Equal(t, http.StatusOK, response.Code)
+	})
+	t.Run("AccessControl allows viewing CurrentOrgQuotas with exact permissions", func(t *testing.T) {
+		setAccessControlPermissions(acmock, []*accesscontrol.Permission{{Action: ActionOrgsRead, Scope: accesscontrol.Scope("orgs", "1")}})
+		response := callAPI(server, http.MethodGet, getCurrentOrgQuotasUrl, nil, t)
+		assert.Equal(t, http.StatusOK, response.Code)
+	})
+	t.Run("AccessControl prevents viewing CurrentOrgQuotas with incorrect permissions", func(t *testing.T) {
+		setAccessControlPermissions(acmock, []*accesscontrol.Permission{{Action: "orgs:invalid"}})
+		response := callAPI(server, http.MethodGet, getCurrentOrgQuotasUrl, nil, t)
+		assert.Equal(t, http.StatusForbidden, response.Code)
+	})
+}
+
+func TestAPIEndpoint_GetOrgQuotas_LegacyAccessControl(t *testing.T) {
+	testuser := &models.SignedInUser{UserId: testUserID, OrgId: 1, OrgRole: models.ROLE_ADMIN, Login: testUserLogin, IsGrafanaAdmin: true}
+	server, hs, _ := setupHTTPServer(t, false, testuser)
+
+	hs.Cfg.Quota.Enabled = true
+	hs.Cfg.Quota.Org = &testOrgQuota
+	// Required while sqlstore quota.go relies on setting global variables
+	setting.Quota = hs.Cfg.Quota
+
+	// Create two orgs, to fetch another one than the logged in one
+	_, err := hs.SQLStore.CreateOrgWithMember("TestOrg", testUserID)
+	require.NoError(t, err)
+	_, err = hs.SQLStore.CreateOrgWithMember("TestOrg2", testUserID)
+	require.NoError(t, err)
+
+	t.Run("Admin can view another org quotas", func(t *testing.T) {
+		response := callAPI(server, http.MethodGet, fmt.Sprintf(getOrgsQuotasUrl, 2), nil, t)
+		assert.Equal(t, http.StatusOK, response.Code)
+	})
+}
+
+func TestAPIEndpoint_GetOrgQuotas_AccessControl(t *testing.T) {
+	testuser := &models.SignedInUser{UserId: testUserID, OrgId: 1, OrgRole: models.ROLE_VIEWER, Login: testUserLogin}
+	server, hs, acmock := setupHTTPServer(t, true, testuser)
+
+	hs.Cfg.Quota.Enabled = true
+	hs.Cfg.Quota.Org = &testOrgQuota
+	// Required while sqlstore quota.go relies on setting global variables
+	setting.Quota = hs.Cfg.Quota
+
+	// Create two orgs, to fetch another one than the logged in one
+	_, err := hs.SQLStore.CreateOrgWithMember("TestOrg", testUserID)
+	require.NoError(t, err)
+	_, err = hs.SQLStore.CreateOrgWithMember("TestOrg2", testUserID)
+	require.NoError(t, err)
+
+	t.Run("AccessControl allows viewing another org quotas with correct permissions", func(t *testing.T) {
```

Comment:
Might be worth adding a test with exact permissions here as well (read only org 2).